includes/razorpay-subscription-webhook.php

```diff
@@ -199,8 +202,43 @@ protected function processSubscriptionSuccess($orderId, $subscription, $paymentI
      *
      * @param $orderId
      */
-    protected function processSubscriptionFailed($orderId)
+    protected function processSubscriptionFailed($orderId, $subscription, $paymentId)
     {
-        WC_Subscriptions_Manager::process_subscription_payment_failure_on_order($orderId);
+        $wcSubscription = wcs_get_subscriptions_for_order($orderId);
+        $wcSubscription = array_values($wcSubscription)[0];
+
+        $is_first_payment = ( $wcSubscription->get_completed_payment_count() < 1 ) ? true : false;
+
+        if (!$is_first_payment)
+        {
+            if ( $wcSubscription->has_status( 'active' ) )
+            {
+                $wcSubscription->update_status( 'on-hold' );
+            }
+
+            $renewal_order = $this->get_renewal_order_by_transaction_id( $wcSubscription, $paymentId );
+
+            if ( is_null( $renewal_order ) ) {
+                $renewal_order = wcs_create_renewal_order( $wcSubscription );
+            }
+
+            $available_gateways = WC()->payment_gateways->get_available_payment_gateways();
+            $renewal_order->set_payment_method( $available_gateways['razorpay_subscriptions'] );
```

Comment:
available_gateways  would be hash.
`array_keys($available_gateways))
 ["cod","paypal","razorpay"]
`